server/api/impl/proxy/asset.go

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2020-present Mattermost, Inc. All Rights Reserved.
+// See License for license information.
+
+package proxy
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/pkg/errors"
+
+	"github.com/mattermost/mattermost-plugin-apps/apps"
+)
+
+func (p *Proxy) GetAsset(appID apps.AppID, assetName string) ([]byte, error) {
+	app, err := p.store.App().Get(appID)
+	if err != nil {
+		return nil, errors.Wrapf(err, "can't load app - %s", appID)
+	}
+	for _, asset := range app.Manifest.Assets {
+		if asset.Name != assetName {
+			continue
+		}
+		errorMessage := fmt.Sprintf("can't download %s for appID - %s, assetName - %s from", asset.Type, appID, assetName)
+		switch asset.Type {
+		case apps.S3Asset:
+			data, err := p.awsClient.S3FileDownload(asset.Bucket, asset.Key)
+			if err != nil {
+				return nil, errors.Wrapf(err, "%s %s/%s", errorMessage, asset.Bucket, asset.Key)
+			}
+			return data, nil
+		case apps.HTTPAsset:
+			/* #nosec G107 */
```

Comment:
with HTTP we should pipe from resp.Body into the incoming' response, not return the bytes. Ideally ditto for S3. A separate (performance) ticket?

Also, size limit?